plugin-script-python/src/main/java/io/kestra/plugin/scripts/python/AbstractPythonExecScript.java

```diff
@@ -0,0 +1,149 @@
+    @Schema(
+        title = "The version of Python to use for the script.",
+        description = "If no version is explicitly specified, the task will attempt to extract the version from the configured container image. If it cannot determine the version from the image, the task will default to Python '"+ DEFAULT_IMAGE +" '"
+    )
+    @PluginProperty
+    protected Property<String> pythonVersion;
+
+    @Schema(
+        title = "Enable Python dependency caching",
+        description = "When enabled, Python dependencies will be cached across task executions. This locks dependency versions and speeds up subsequent runs by avoiding redundant installations."
+    )
+    @PluginProperty
+    protected Property<Boolean> dependencyCacheEnabled = Property.of(true);
+
+    protected ResolvedPythonEnvironment setupPythonEnvironment(final RunContext runContext, final boolean isCacheEnabled) throws IllegalVariableEvaluationException, IOException {
+        List<String> requirements = new ArrayList<>(runContext.render(dependencies).asList(String.class));
+        DEFAULT_PYTHON_PACKAGES.forEach(pkg -> addPackageIfNoneMatch(requirements, pkg));
+
+        final String targetPythonVersion = getTargetPythonVersion(runContext);
+
+        final Path localCacheDir = ((DefaultRunContext) runContext).getApplicationContext().getEnvironment().getProperty("kestra.cache.dir", String.class)
+            .map(Path::of)
+            .orElse(Path.of("/tmp/kestra/cache"));
+
+        final PythonDependenciesResolver resolver = new PythonDependenciesResolver(runContext.logger(), runContext.workingDir(), localCacheDir);
+
+        final String hash = resolver.getRequirementsHashKey(targetPythonVersion, requirements);
+
+        final long metricCacheDownloadStart = System.currentTimeMillis();
+
+        Optional<InputStream> cacheFile = isCacheEnabled ? runContext.storage().getCacheFile("python-" + getType(), hash) : Optional.empty();
+
+        final ResolvedPythonPackages resolvedPythonPackages;
+        final boolean cached;
+        if (cacheFile.isPresent()) {
+            runContext.logger().debug("Restoring python dependencies cache for key: {}", hash);
+            resolvedPythonPackages = resolver.getPythonLibs(targetPythonVersion, hash, cacheFile.get());
+            runContext.logger().debug("Cache restored successfully");
+            runContext.metric(Timer.of("cache.download.duration", Duration.ofMillis(System.currentTimeMillis() - metricCacheDownloadStart)));
+            cached = true;
+        } else {
+            if (isCacheEnabled) {
+                runContext.logger().debug("Could not find python dependencies cache for key: {}", hash);
+            }
+            resolvedPythonPackages = resolver.getPythonLibs(targetPythonVersion, hash, requirements);
+            cached = false;
+        }
+
+        runContext.logger().debug("Installed dependencies: {}", resolvedPythonPackages.packagesToString());
+
+        String pythonInterpreter = "python";
```

Comment:
The interpreter can be configured in the Python task itself so better use it from there.
Example usage: python3 instead of python :(

Reply:
Actually, the  `python` interepreter is already hard-coded right in 0.22x version. It's only on the legacy Python Task that the user could configure it. So this PR doesn't really change the behavior.